core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineSqlite.java

```diff
@@ -0,0 +1,248 @@
+    return !tableName.contains(NAMESPACE_SEPARATOR);
+  }
+
+  @Override
+  public String computeBooleanValue(boolean value) {
+    return value ? "TRUE" : "FALSE";
+  }
+
+  @Override
+  public String[] createNamespaceSqls(String fullNamespace) {
+    // In SQLite storage, namespace will be added to table names as prefix along with underscore
+    // separator.
+    return new String[0];
+  }
+
+  /**
+   * @param hasDescClusteringOrder Ignored. SQLite cannot handle key order.
+   * @see <a href="https://www.sqlite.org/syntax/table-constraint.html">table-constraint</a>
+   */
+  @Override
+  public String createTableInternalPrimaryKeyClause(
+      boolean hasDescClusteringOrder, TableMetadata metadata) {
+    return "PRIMARY KEY ("
+        + Stream.concat(
+                metadata.getPartitionKeyNames().stream(), metadata.getClusteringKeyNames().stream())
+            .map(this::enclose)
+            .collect(Collectors.joining(","))
+        + "))";
+  }
+
+  @Override
+  public String[] createTableInternalSqlsAfterCreateTable(
+      boolean hasDescClusteringOrder, String schema, String table, TableMetadata metadata) {
+    // do nothing
+    return new String[] {};
+  }
+
+  @Override
+  public String tryAddIfNotExistsToCreateTableSql(String createTableSql) {
+    return createTableSql.replace("CREATE TABLE", "CREATE TABLE IF NOT EXISTS");
+  }
+
+  @Override
+  public String[] createMetadataSchemaIfNotExistsSql(String metadataSchema) {
+    // Do nothing. Namespace is just a table prefix in the SQLite implementation.
+    return new String[0];
+  }
+
+  @Override
+  public boolean isCreateMetadataSchemaDuplicateSchemaError(SQLException e) {
+    // Namespace is never created
+    return false;
+  }
+
+  @Override
+  public String deleteMetadataSchemaSql(String metadataSchema) {
+    // Do nothing. Metadata schema is just a prefix to the metadata table in the SQLite
+    // implementation.
+    return null;
+  }
+
+  @Override
+  public String dropNamespaceSql(String namespace) {
+    // Do nothing. Namespace is just a table prefix in the SQLite implementation.
+    return null;
+  }
+
+  @Override
+  public void dropNamespaceTranslateSQLException(SQLException e, String namespace)
+      throws ExecutionException {
+    throw new AssertionError("dropNamespace never happen in SQLite implementation");
+  }
+
+  @Override
+  public String namespaceExistsStatement() {
+    return "SELECT 1 FROM sqlite_master WHERE "
+        + enclose("type")
+        + " = \"table\" AND "
+        + enclose("tbl_name")
+        + " LIKE ?";
+  }
+
+  @Override
+  public String namespaceExistsPlaceholder(String namespace) {
+    return namespace + NAMESPACE_SEPARATOR + "%";
+  }
+
+  @Override
+  public String alterColumnTypeSql(
+      String namespace, String table, String columnName, String columnType) {
+    throw new AssertionError(
+        "SQLite does not require changes in column data types when making indices.");
```

Comment:
> Maybe, we should rename it

OK, I will do it in this PR.

> since it's part of the public interface.

Actually, I don't want to make `interface RdbEngineStrategy` public.
It should not be implemented outside of the scalardb projects to avoid backward-compatibility considerations.

It's good if we can give `pub(crate)`-like visibility in Rust.
https://doc.rust-lang.org/reference/visibility-and-privacy.html#pubin-path-pubcrate-pubsuper-and-pubself

Do we have any ways to do such things in Java?

> we need documentation/comments in RdbEngineStrategy to describe what each interface is supposed to do.

Anyway, I agree with it. I'll do it in another PR.